packages/teleport/src/components/NodeList/NodeList.tsx

```diff
@@ -16,126 +16,64 @@ limitations under the License.
 
 import React from 'react';
 import styled from 'styled-components';
-import { sortBy } from 'lodash';
-import isMatch from 'design/utils/match';
-import {
-  Column,
-  SortHeaderCell,
-  Cell,
-  TextCell,
-  SortTypes,
-  renderLabelCell,
-} from 'design/DataTable';
-import Table from 'design/DataTable/Paged';
+import Table, { Cell, LabelCell } from 'design/DataTableNext';
 import MenuSshLogin, { LoginItem } from 'shared/components/MenuSshLogin';
 import { Node } from 'teleport/services/nodes';
 
 function NodeList(props: Props) {
-  const {
-    nodes = [],
-    search,
-    onSearchChange,
-    onLoginMenuOpen,
-    onLoginSelect,
-    pageSize = 100,
-  } = props;
-  const [sortDir, setSortDir] = React.useState<Record<string, string>>({
-    hostname: SortTypes.DESC,
-  });
-
-  function sortAndFilter(search) {
-    const filtered = nodes.filter(obj =>
-      isMatch(obj, search, {
-        searchableProps: ['hostname', 'addr', 'tags', 'tunnel'],
-        cb: searchAndFilterCb,
-      })
-    );
-
-    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
-    const sorted = sortBy(filtered, columnKey);
-    if (sortDir[columnKey] === SortTypes.ASC) {
-      return sorted.reverse();
-    }
-
-    return sorted;
-  }
-
-  function onSortChange(columnKey: string, sortDir: string) {
-    setSortDir({ [columnKey]: sortDir });
-  }
-
-  const data = sortAndFilter(search);
+  const { nodes = [], onLoginMenuOpen, onLoginSelect, pageSize = 100 } = props;
 
   return (
-    <div>
-      <StyledTable
-        pageSize={pageSize}
-        data={data}
-        search={search}
-        onSearchChange={onSearchChange}
-      >
-        <Column
-          columnKey="hostname"
-          header={
-            <SortHeaderCell
-              sortDir={sortDir.hostname}
-              onSortChange={onSortChange}
-              title="Hostname"
-            />
-          }
-          cell={<TextCell />}
-        />
-        <Column
-          columnKey="addr"
-          header={
-            <SortHeaderCell
-              sortDir={sortDir.addr}
-              onSortChange={onSortChange}
-              title="Address"
+    <StyledTable<Node>
```

Comment:
Can we try to infer this type from the `data` prop so we do not need to explicitly specify it?